src/main.ts

```diff
@@ -154,6 +169,18 @@ function generateHandleServerProviderEvent() {
 	};
 }
 
+function installPostgreSQLProjectTemplate(dotNetSdk: DotNetInfo, templateNupkg: string): Promise<void> {
+	return new Promise<void>((_resolve, _reject) => {
+		let args = ['new', '-i', templateNupkg];
+		runDotNetCommand(dotNetSdk, args).then(() => {
+			Telemetry.sendTelemetryEvent('Project template installed.');
+		}, e => {
+			Telemetry.sendTelemetryEvent('ProjectTemplateInstallationFailed, error: ${error}', e.message);
+			vscode.window.showErrorMessage('Failed to install project template');
+		});
+	});
+}
+
 // this method is called when your extension is deactivated
 export function deactivate(): void {
```

Comment:
This is just a deactivate function which gets executed on vscode shutdown. I ll check the uninstall.

Reply:
So they have an uninstall hook where you can specify a Node.js script to run.  You set this in package.json Scripts section like this:
"vscode:uninstall": "node ./out/extensionUninstall"
I was able to add this and uninstall the template but the problem is that at this point the script would not have access to the vscode api and therefore I cannot really preserve if this template was installed by us. As we cannot just remove it if user had it already.
Trying to see what we can do.

Reply:
How would you check, if it was installed only by us? I don't think there is a reliable way to do ref-counting... It might be also installed by an extension in vscode itself, not just ADS, if/when we integrate there. That's typical problem with shared installations.
Is there a way to prompt the user somehow? If you ask me - I think it's better to uninstall it and let them install it back, if they still need it or not.
Alternatively, how often do we need the template? Is it only to create new projects? How long does the installation take? Can we make installation part of the new project command - install, create project, uninstall? Then you can check, if it's already installed and only uninstall it, if we installed it ourselves.

Reply:
It is already part of new project command, we only install the template if it is not there when user creates a new project. The only prpblem is to figure out if we installed it during install. We can do this by creating a hacky solution where we store this info somewhere. But there is no straightforward way as you lose all vscode extension info by then.